src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import { TempTodo, Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo | TempTodo;
+  onDelete?: (id: number) => void;
+  handleStatusChange?: (todo: Todo) => void
+  updatingTodo?: boolean;
+  editTodo?: (todo: Todo, newTitle:string) => void;
+};
+export const TodoInfo:React.FC<Props> = React.memo(({
+  onDelete,
+  todo,
+  handleStatusChange,
+  updatingTodo,
+  editTodo,
+}) => {
+  // eslint-disable-next-line no-console
+  console.log('rendering todoinfo');
+  const [isEditing, setEditing] = useState(false);
+  const [title, setTitle] = useState(todo.title);
+
+  const isLoaderActive = todo.id === 0
+        || updatingTodo;
+
+  const handleEdit = () => {
+    if (!title && onDelete) {
+      onDelete(todo.id);
+    }
+
+    if (editTodo) {
+      editTodo(todo, title);
+      setEditing(false);
+    }
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classnames('todo',
+        { completed: todo.completed })}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={() => {
+            if (handleStatusChange) {
```

Comment:
I know that's because of `TempTodo`, but I guess it's better to create a function like this `const doNothing = () => {};` above component and use it as a default value for all functions you need instead of checking it every time